.github/workflows/gitlab-ci.yaml

```
@@ -0,0 +1,86 @@
+    - docker:dind
+  before_script:
+    - docker login -u "$CI_REGISTRY_USER" -p "$CI_REGISTRY_PASSWORD" $CI_REGISTRY
+    - export IMAGE_TAG="$CI_REGISTRY_IMAGE:$TEST_IMAGE_TAG"
+  script:
+    - docker build -t $IMAGE_TAG .
+    - docker push $IMAGE_TAG
+  rules:
+    - if: $CI_COMMIT_BRANCH != "main"
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+
+test:
```

actionlint (1.7.4)

unexpected key "test" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name"

(syntax-check)